### PR TITLE
feat: added a detailed PR Template

### PR DESCRIPTION
**Pull request by MrKrishnaAgarwal**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- Feature, Docs

- **Why was this change needed?** (You can also link to an open issue here)
- This change was needed because I think a detailed PR template would be better and is a good option both for maintainers and contributors

- **Other information**:
- Any suggestions and improvements are welcome!
cc @scopsy

### CODE REVIEW

**Comment by MrKrishnaAgarwal** [comment]
Preview:
image

**Comment by MrKrishnaAgarwal** [comment]
After the user fills, example:
image

**Comment by MrKrishnaAgarwal** [comment]
**I think it's a good addition, rest up to you :)**

**Comment by davidsoderberg** [comment]
Hi @MrKrishnaAgarwal Thanks for the PR :
We have discussed this topic before inside of Novu and our thinking is that eslint, prettier and Github actions should be able to fix/answer the questions you have suggested 😄 

What I think would be good to add is maybe some text under the "other information" heading we have today is that we would  appreciate if screenshots was provided of UI changes, what do you think @MrKrishnaAgarwal, @ainouzgali @p-fernandez @djabarovgeorge @BiswaViraj ?

**Comment by MrKrishnaAgarwal** [comment]
> Hi @MrKrishnaAgarwal Thanks for the PR :
> We have discussed this topic before inside of Novu and our thinking is that eslint, prettier and Github actions should be able to fix/answer the questions you have suggested 😄 
> 
> What I think would be good to add is maybe some text under the "other information" heading we have today is that we would  appreciate if screenshots was provided of UI changes, what do you think @MrKrishnaAgarwal, @ainouzgali @p-fernandez @djabarovgeorge @BiswaViraj ?

Sure, I can add that point there, how's the PR Template overall?

**Comment by MrKrishnaAgarwal** [comment]
and I think the UI changes can be added by the user under the Screenshots section

**Comment by davidsoderberg** [comment]
> > Hi @MrKrishnaAgarwal Thanks for the PR :
> > We have discussed this topic before inside of Novu and our thinking is that eslint, prettier and Github actions should be able to fix/answer the questions you have suggested 😄
> 
> Sure, I can add that point there, how's the PR Template overall?

I think most of it is not needed because the stuff mentioned in my first comment but we might add some text in the current "other information" section about screenshots 😄

**Comment by MrKrishnaAgarwal** [comment]
> > > Hi @MrKrishnaAgarwal Thanks for the PR :
> > > We have discussed this topic before inside of Novu and our thinking is that eslint, prettier and Github actions should be able to fix/answer the questions you have suggested 😄
> > 
> > Sure, I can add that point there, how's the PR Template overall?
> 
> I think most of it is not needed because the stuff mentioned in my first comment but we might add some text in the current "other information" section about screenshots 😄 

Ohh okay
So should I close the pr

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,5 +1,49 @@
...
+-->
 
-- **Why was this change needed?** (You can also link to an open issue here)
+### What kind of change does this PR introduce? <!-- Mark The Boxes -->
 
-- **Other information**:
+- [ ] Bug
+- [ ] Feature
+- [ ] Docs
+- [ ] Other(s)
```
Comment (ainouzgali) [inline .github/PULL_REQUEST_TEMPLATE.md:12]: I like this addition! It will help make it clearer for people who open a pr that they need to mention it and for us to immediately see as it will be in the same format all the time

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,5 +1,49 @@
...
+<!-- Explain all the changes in the PR and tell what is the use of this change. -->
+
+<!-- Mark all the applicable boxes. To mark the box as done follow the following conventions -->
+<!--
+[x] - Correct; marked as done
+[X] - Correct; marked as done
+[ ] - Not correct; marked as **not** done
+-->
+
+### Check List (Check all the applicable boxes) <!-- Follow the above conventions to check the box -->
```
Comment (ainouzgali) [inline .github/PULL_REQUEST_TEMPLATE.md:34]: I agree with David, that this checklist is not needed at the moment

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (davidsoderberg) [reply .github/PULL_REQUEST_TEMPLATE.md:12]: Yeah, I guess this could make this question easier to answer "What kind of change does this PR introduce?" 😄

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (MrKrishnaAgarwal) [reply .github/PULL_REQUEST_TEMPLATE.md:34]: I'll fix it and remove it

Comment (MrKrishnaAgarwal) [reply .github/PULL_REQUEST_TEMPLATE.md]: I have removed it @ainouzgali

**Review by p-fernandez — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,5 +1,36 @@
...
-- **Other information**:
+- [ ] Bug
+- [ ] Feature
+- [ ] Docs
+- [ ] Other(s)
+
+
+<!-- If your PR fixes an open issue, use `Closes #999` to link your PR with the issue. #999 stands for the issue number you are fixing -->
+
+### Fixes Issue
```
Comment (davidsoderberg) [inline .github/PULL_REQUEST_TEMPLATE.md:18]: I think this was not included in the old template? 
There is not not always an issue so I would rather not see this heading at all, could be included in "Why was this change needed"
@ainouzgali @p-fernandez what do you think?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -1,5 +1,36 @@
...
+<!-- Remove this section if not applicable -->
+
+<!-- Example: Closes #31 -->
+
+### Why was this change needed?
+
+<!-- Explain all the changes in the PR and tell what is the use of this change. -->
+
+
+### Screenshots
```
Comment (davidsoderberg) [inline .github/PULL_REQUEST_TEMPLATE.md:29]: This text could be moved into "Other information"

Comment (MrKrishnaAgarwal) [reply .github/PULL_REQUEST_TEMPLATE.md:29]: No, I think Screenshots should be a different heading only so that's easy
What to other think @scopsy

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (MrKrishnaAgarwal) [reply .github/PULL_REQUEST_TEMPLATE.md:18]: Let me know what others think about it, and I'll remove if more agree.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (davidsoderberg) [reply .github/PULL_REQUEST_TEMPLATE.md:29]: Screenshots are only needed for UI changes and because of that I think it belongs in "Other information"

Comment (MrKrishnaAgarwal) [reply .github/PULL_REQUEST_TEMPLATE.md:29]: Okay!
Should I do the changes then @davidsoberg

Comment (p-fernandez) [reply .github/PULL_REQUEST_TEMPLATE.md:29]: @MrKrishnaAgarwal please follow @davidsoderberg suggestion.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (p-fernandez) [reply .github/PULL_REQUEST_TEMPLATE.md:18]: @MrKrishnaAgarwal please follow @davidsoderberg suggestion.

**Review by p-fernandez — CHANGES_REQUESTED** [review]
Please address @davidsoderberg comments.

**Comment by MrKrishnaAgarwal** [comment]
> Please address @davidsoderberg comments.

Okay Sure!

**Comment by MrKrishnaAgarwal** [comment]
@davidsoderberg @p-fernandez I have resolved all the things and updated accordingly.
Please review now because I think it is `Ready-To-Merge` now, any other improvements are welcome!

**Review by p-fernandez — APPROVED** [review]

**Review by scopsy — APPROVED** [review]

**Review by scopsy — APPROVED** [review]

**Comment by MrKrishnaAgarwal** [comment]
@scopsy can you approve the checks

**Comment by MrKrishnaAgarwal** [comment]
@scopsy
Please approve the checks so that I can see that they are passing, and merge the PR!

**Comment by MrKrishnaAgarwal** [comment]
@davidsoderberg and @scopsy Please merge the PR if you think it's fine.

**Comment by MrKrishnaAgarwal** [comment]
@ainouzgali and @scopsy Can you review the PR and merge it!

**Comment by MrKrishnaAgarwal** [comment]
thankyou @ainouzgali ✨
